fuzz/fuzz_targets/fuzz_int_set.rs

```diff
@@ -0,0 +1,673 @@
+#![no_main]
+//! A correctness fuzzer that checks all of the public API methods of IntSet.
+//!
+//! This fuzzer exercises the public API of IntSet and compares the results to the same operations run
+//! on a BTreeSet. Any differences in behaviour and/or set contents triggers a panic.
+//!
+//! The fuzzer input data is interpretted as a series of op codes which map to the public api methods of IntSet.
+//!
+//! Note: currently only inclusive mode IntSet's are tested.
+
+use std::cmp::max;
+use std::ops::Bound::Excluded;
+use std::ops::Bound::Included;
+use std::ops::RangeInclusive;
+use std::{collections::BTreeSet, error::Error};
+
+use int_set::IntSet;
+use libfuzzer_sys::fuzz_target;
+
+const OPERATION_COUNT: usize = 7_500;
+
+// TODO(garretrieger): allow inverted sets to be accessed.
+// TODO(garretrieger): allow a limited domain set to be accessed.
+
+struct Input<'a> {
+    // The state includes 2 of each type of sets to allow us to test out binary set operations (eg. union)
+    int_set: &'a mut IntSet<u32>,
+    btree_set: &'a mut BTreeSet<u32>,
+}
+
+impl Input<'_> {
+    fn from<'a>(int_set: &'a mut IntSet<u32>, btree_set: &'a mut BTreeSet<u32>) -> Input<'a> {
+        Input { int_set, btree_set }
+    }
+}
+
+trait Operation {
```

Comment:
It seems to me your life would be easier if Operation was an enum that defined size and operation in it's impl. It would then have a fixed, small, size and you wouldn't need the whole mess of boxed dyn traits.

Reply:
If this is done as an enum then every time a new type is added 3 match statements will need to be updated, versus the current setup where only one (op code -> constructor) needs to be updated. The mild performance improvement isn't important in this context (the performed set operations are going to significantly outweigh the cost of dyn here) so the advantages of enums reducing size aren't relevant.

Reply:
My initial write of this was using enum's but I switched to dyn because I found enums to be cumbersome when adding new ops.